In [162]:
from scipy import stats

**Задача № 1**

    Случайная непрерывная величина A имеет равномерное распределение на
    промежутке (200, 800].
    Найдите ее среднее значение и дисперсию.

In [163]:
a,b = 200, 800

In [164]:
# Среднее значение

M = (a + b)/2
M

500.0

In [165]:
#Дисперсия

D = ((b - a)**2)/12
D

30000.0

**Задача №2** 

    О случайной непрерывной равномерно распределенной величине B известно, что ее
    дисперсия равна 0.2.
    Можно ли найти правую границу величины B и ее среднее значение зная, что левая
    граница равна 0.5?
    Если да, найдите ее.

In [166]:
D, a = 0.2, 0.5

(b - a)^2 / 12 = D

(b - 0.5)^2 = 0.2 * 12 = 2.4

b1 = sqrt(2.4) + 0.5

b2 = - sqrt(2.4) + 0.5

In [167]:
b1 = round(2.4**(1/2) + 0.5,2)
b1

2.05

In [168]:
b2 = -2.4**1/2 + 0.5
b2

-0.7

Так как левая граница величины В - положительная, то подходит только корень b1. Значит правая граница существует и равна 2.05

In [169]:
#Среднее значение

M = round((a + b1)/2,2)
M

1.27

**Задача №3**

    Непрерывная случайная величина X распределена нормально и задана плотностью
    распределения
    f(x) = (1 / (4 * sqrt(2pi))) * exp((-(x+2)**2) / 32)
    Найдите:
    а). M(X)
    б). D(X)
    в). std(X) (среднее квадратичное отклонение)

In [170]:
M = -2

In [171]:
D = 4**2
D

16

In [172]:
sko = 4

**Задача №4**

Рост взрослого населения города X имеет нормальное распределение, причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
1. больше 182 см?
2. больше 190 см?
3. от 166 см до 190 см?
4. от 166 см до 182 см?
5. от 158 см до 190 см?
6. не выше 150 см или не ниже 190 см?
7. не выше 150 см или не ниже 198 см?
8. ниже 166 см?

Задачу можно решить двумя способами: без использования сторонних библиотек (numpy, scipy, pandas и пр.), а затем проверить себя с помощью встроенных функций

In [173]:
M, sko = 174, 8

**пункт 1.** Вероятность того, что случайным образом выбранный взрослый человек имеет рост больше **182 см**

In [174]:
z = (182 - M) / sko
z

1.0

По таблице находим вероятность по полученной z. Т.к находим больше 182 см, то от 1 отнимаем найденное значение
Аналогичный результат получим по правилу 3-х сигм. 182-174 = 8 - это одна сигма, поэтому вероятность p = (1-0.68)/2 = 0.16

In [175]:
p = 1 - 0.8413 
p

0.15869999999999995

***Проверка***

In [176]:
1 - stats.norm.cdf(182, loc = M, scale = sko)

0.15865525393145707

**пункт 2** Вероятность того, что случайным образом выбранный взрослый человек имеет рост больше **190 см**

In [177]:
z = (190 - M) / sko
z

2.0

По таблице находим вероятность по полученной z. Т.к находим больше 190 см, то от 1 отнимаем найденное значение

In [178]:
p = 1 - 0.9772 
p

0.022800000000000042

***Проверка***

In [179]:
1 - stats.norm.cdf(190, loc = M, scale = sko)

0.02275013194817921

**пункт 3** Вероятность того, что случайным образом выбранный взрослый человек имеет рост **от 166 см до 190 см**

In [180]:
z1 = (166 - M) / sko
z1

-1.0

In [181]:
p = 0.9772 - 0.1587
p

0.8185

***Проверка***

In [182]:
stats.norm.cdf(190, loc = M, scale = sko) - stats.norm.cdf(166, loc = M, scale = sko) 

0.8185946141203637

**пункт 4** Вероятность того, что случайным образом выбранный взрослый человек имеет рост от 166 см до 182 см

По правилу 3-х сигм, вероятность p = 0.68

***Проверка***

In [183]:
stats.norm.cdf(182, loc = M, scale = sko) - stats.norm.cdf(166, loc = M, scale = sko) 

0.6826894921370859

**пункт 5** Вероятность того, что случайным образом выбранный взрослый человек имеет рост от 158 см до 190 см

По правилу 3-х сигм, вероятность p = 0.954

***Проверка***

In [184]:
stats.norm.cdf(190, loc = M, scale = sko) - stats.norm.cdf(158, loc = M, scale = sko) 

0.9544997361036416

**пункт 6** Вероятность того, что случайным образом выбранный взрослый человек имеет рост не выше 150 см или не ниже 190 см

In [185]:
z1 = (150 - M) / sko
z1

-3.0

In [186]:
p = 0.0013 + 0.0228
p

0.0241

***Проверка***

In [187]:
(1 - stats.norm.cdf(190, loc = M, scale = sko)) + stats.norm.cdf(150, loc = M, scale = sko) 

0.0241000299798093

**пункт 7** Вероятность того, что случайным образом выбранный взрослый человек имеет рост не выше 150 см или не ниже 198 см

По правилу 3-х сигм, вероятность p = 1 - 0.9972

In [188]:
p = 1 - 0.9972
p

0.0028000000000000247

***Проверка***

In [189]:
(1 - stats.norm.cdf(198, loc = M, scale = sko)) + stats.norm.cdf(150, loc = M, scale = sko) 

0.0026997960632601965

**пункт 8** Вероятность того, что случайным образом выбранный взрослый человек имеет рост ниже 166 см

По правилу 3-х сигм, вероятность p = (1 - 0.68)/2

In [190]:
p = (1 - 0.68) / 2
p

0.15999999999999998

***Проверка***

In [191]:
stats.norm.cdf(166, loc = M, scale = sko)

0.15865525393145707

**Задача №5.** 

На сколько сигм (средних квадратичных отклонений) отклоняется рост человека,
равный 190 см, от
математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

In [192]:
M, D, X = 178, 25, 190

In [193]:
sko = D**(1/2)
sko

5.0

In [194]:
res = (X - M) / sko
res

2.4